projects/drivers/xilinx_platform/platform_drivers.c

```diff
@@ -180,55 +193,62 @@ int32_t spi_init(struct spi_desc **desc,
 		 const struct spi_init_param *param)
 {
 	spi_desc *descriptor;
+	extra_spi_desc *extra_descriptor;
+	extra_spi_init_param *extra_param;
 	int32_t ret;
 
 	descriptor = (struct spi_desc *) calloc(1, sizeof(*descriptor));
-	if (!descriptor)
+	extra_descriptor = (struct extra_spi_desc *) calloc(1,
+			   sizeof(*extra_descriptor));
+	if (!descriptor || !extra_descriptor)
 		return FAILURE;
 
+	descriptor->extra = extra_descriptor;
+	extra_param = param->extra;
+
 	descriptor->mode = param->mode;
 	descriptor->chip_select = param->chip_select;
-	descriptor->flags = param->flags;
+	extra_descriptor->flags = extra_param->flags;
 
 #ifdef _XPARAMETERS_PS_H_
-	descriptor->config = XSpiPs_LookupConfig(param->id);
-	if (descriptor->config == NULL)
+	extra_descriptor->config = XSpiPs_LookupConfig(extra_descriptor->id);
+	if (extra_descriptor->config == NULL)
 		goto error;
 
-	ret = XSpiPs_CfgInitialize(&descriptor->instance,
-				   descriptor->config, descriptor->config->BaseAddress);
+	ret = XSpiPs_CfgInitialize(&extra_descriptor->instance,
+				   extra_descriptor->config, extra_descriptor->config->BaseAddress);
 	if (ret != 0)
 		goto error;
 
-	XSpiPs_SetOptions(&descriptor->instance,
+	XSpiPs_SetOptions(&extra_descriptor->instance,
 			  XSPIPS_MASTER_OPTION |
-			  ((descriptor->flags & SPI_CS_DECODE) ?
+			  ((extra_descriptor->flags & SPI_CS_DECODE) ?
 			   XSPIPS_DECODE_SSELECT_OPTION : 0) |
 			  XSPIPS_FORCE_SSELECT_OPTION |
 			  ((descriptor->mode & SPI_CPOL) ?
 			   XSPIPS_CLK_ACTIVE_LOW_OPTION : 0) |
 			  ((descriptor->mode & SPI_CPHA) ?
 			   XSPIPS_CLK_PHASE_1_OPTION : 0));
 
-	XSpiPs_SetClkPrescaler(&descriptor->instance,
+	XSpiPs_SetClkPrescaler(&extra_descriptor->instance,
 			       XSPIPS_CLK_PRESCALE_64);
 
-	XSpiPs_SetSlaveSelect(&descriptor->instance, 0xf);
+	XSpiPs_SetSlaveSelect(&extra_descriptor->instance, 0xf);
 #else
-	ret = XSpi_Initialize(&descriptor->instance, param->id);
+	ret = XSpi_Initialize(&extra_descriptor->instance, extra_param->id);
```

Comment:
in some places you use extra_param->id and in others extra_descriptor->id. 

This is happening because of the extra variable you don't actually need, and you end up with inconsistencies.